components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/java/org/wso2/carbon/user/mgt/ui/Util.java

```diff
@@ -198,8 +210,63 @@ public static void updateCheckboxStateMap(Map<String, Boolean> checkBoxMap, Map<
         }
     }
 
-    public static boolean isAskPasswordEnabled(){
+    public static boolean isAskPasswordEnabled() {
         return isAskPasswordEnabled;
     }
 
+    public static boolean isUserOnBoardingEnabled(ServletContext context, HttpSession session) throws Exception {
+        Object emailVerificationEnabledObj = session.getAttribute(EMAIL_VERIFICATION_ENABLE_PROP_NAME);
+        if (emailVerificationEnabledObj != null) {
+            return Boolean.parseBoolean(String.valueOf(emailVerificationEnabledObj));
+        }
+        String backendServerURL = CarbonUIUtil.getServerURL(context, session);
+        String cookie = (String) session.getAttribute(ServerConstants.ADMIN_SERVICE_COOKIE);
+        ConfigurationContext configContext =
+                (ConfigurationContext) context.getAttribute(CarbonConstants.CONFIGURATION_CONTEXT);
+        Map<String, Map<String, List<ConnectorConfig>>> connectorList;
+        try {
+            IdentityGovernanceAdminClient client =
+                    new IdentityGovernanceAdminClient(cookie, backendServerURL, configContext);
+            connectorList = client.getConnectorList();
+        } catch (RemoteException e) {
+            throw new Exception("Error while calling governance service", e);
+        } catch (IdentityGovernanceAdminServiceIdentityGovernanceExceptionException e) {
+            throw new Exception("Error while calling governance service", e);
+        }
+
+        if (connectorList != null) {
+            for (String key : connectorList.keySet()) {
+                Map<String, List<ConnectorConfig>> subCatList = connectorList.get(key);
+                for (String subCatKey : subCatList.keySet()) {
+                    List<ConnectorConfig> connectorConfigs = subCatList.get(subCatKey);
+                    for (ConnectorConfig connectorConfig : connectorConfigs) {
+                        Property[] properties = connectorConfig.getProperties();
+                        for (Property property : properties) {
+                            if (property.getName().equals(EMAIL_VERIFICATION_ENABLE_PROP_NAME)) {
+                                String propValue = property.getValue();
+                                boolean isEmailVerificationEnabled = false;
+                                if (!StringUtils.isEmpty(propValue)) {
+                                    isEmailVerificationEnabled = Boolean.parseBoolean(propValue);
+                                    session.setAttribute(EMAIL_VERIFICATION_ENABLE_PROP_NAME,
+                                                         isEmailVerificationEnabled);
+                                }
+                                return isEmailVerificationEnabled;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+
+    public static Boolean getUserOnBoarding(HttpSession session) {
```

Comment:
Why we have two similar looking methods?
public static boolean isUserOnBoardingEnabled(ServletContext context, HttpSession session) and 
public static Boolean getUserOnBoarding(HttpSession session)